SimTracker/TrackTriggerAssociation/src/TTTrackAssociationMap.cc

```diff
@@ -9,34 +9,32 @@
 
 /// Operations
 template <>
-edm::Ptr<TrackingParticle> TTTrackAssociationMap<Ref_Phase2TrackerDigi_>::findTrackingParticlePtr(
-    edm::Ptr<TTTrack<Ref_Phase2TrackerDigi_> > aTrack) const {
+TrackingParticlePtr TTTrackAssociationMap<Ref_Phase2TrackerDigi_>::findTrackingParticlePtr(TTTrackPtr aTrack) const {
   if (trackToTrackingParticleMap.find(aTrack) != trackToTrackingParticleMap.end()) {
     return trackToTrackingParticleMap.find(aTrack)->second;
   }
 
   /// Default: return NULL
-  edm::Ptr<TrackingParticle>* temp = new edm::Ptr<TrackingParticle>();
+  TrackingParticlePtr* temp = new TrackingParticlePtr();
   return *temp;
 }
 
 template <>
-std::vector<edm::Ptr<TTTrack<Ref_Phase2TrackerDigi_> > > TTTrackAssociationMap<Ref_Phase2TrackerDigi_>::findTTTrackPtrs(
-    edm::Ptr<TrackingParticle> aTrackingParticle) const {
+std::vector<TTTrackPtr> TTTrackAssociationMap<Ref_Phase2TrackerDigi_>::findTTTrackPtrs(
```

Comment:
this is a suboptimal pattern (return by value -> always creates a copy). instead, return by const ref should be preferred.

to satisfy the need to return an empty vector if nothing is found in the map, a `static const std::vector<...>` can be returned.

ideally, this pattern should be applied to all the `find...()` functions in all of the classes in this package.

Reply:
Nice idea! Done.